pipelines/matrix/src/matrix/pipelines/run_comparison/nodes.py

```diff
@@ -0,0 +1,172 @@
+import logging
+from collections.abc import Callable
+from dataclasses import asdict
+
+import matplotlib.pyplot as plt
+import polars as pl
+from matrix_inject.inject import inject_object
+
+from .evaluations import ComparisonEvaluation
+from .input_paths import InputPathsMultiFold
+from .matrix_pairs import (
+    check_base_matrices_consistent,
+    check_matrix_pairs_equal,
+    give_matrix_pairs_from_lazyframe,
+    harmonize_matrix_pairs,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@inject_object()
+def process_input_filepaths(
+    input_paths: list[InputPathsMultiFold],
+) -> list[dict]:
+    """Function to create input matrices dataset."""
+    # Return initialized dataclass objects as dictionaries
+    return [asdict(v) for v in input_paths]
+
+
+def combine_matrix_pairs(
+    input_matrices: dict[str, any],
+    available_ground_truth_cols: list[str],
+    perform_multifold: bool,
+    assert_data_consistency: bool,
+) -> tuple[pl.LazyFrame, dict[str, any]]:
+    """Function to combine matrix pairs for all folds and models.
+
+    Performs data consistency check or matrix harmonization (restrict to common elements for consistent evaluation) as requested.
+
+    Args:
+        input_matrices: Dictionary containing predictions for all models and folds.
+        available_ground_truth_cols: List of available ground truth columns.
+        perform_multifold: Whether to perform multifold uncertainty estimation. If False, only the first fold of data is used.
```

Comment:
Don't we know based on the input datasets whether the input is multi fold or not? What would be the reason for hooking up a multifold dataset and still setting multifold to false here?